bin/run-markdownlint-tests.sh

```diff
@@ -5,8 +5,15 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 MARKDOWN_TEST_FILES=$(find "${SCRIPT_DIR}/../tests/markdownlint" -type f -name '*.md' | sort)
 
 TEST_DID_FAIL=0
+
+# markdownlint-cli2 config files are merged with other versions of this file in a parent
+# directory, up to the current directory. To prevent the test-specific config from being
+# merged with the project's general config, we have to `cd` to the test directory.
+# Docs: https://github.com/DavidAnson/markdownlint-cli2/tree/main#markdownlint-cli2jsonc
+pushd "${SCRIPT_DIR}/../tests/markdownlint" > /dev/null
+
 for TEST_FILE in $MARKDOWN_TEST_FILES; do
-   OUTPUT=$("${SCRIPT_DIR}/../node_modules/.bin/markdownlint" -c "${SCRIPT_DIR}/../.markdownlint.json" "${TEST_FILE}" 2>&1)
+   OUTPUT=$("${SCRIPT_DIR}/../node_modules/.bin/markdownlint-cli2-config" "${SCRIPT_DIR}/../.markdownlint-cli2.cjs" "${TEST_FILE}" 2>&1)
```

Comment:
Don't we want to use `tests/markdownlint/.markdownlint-cli2.cjs` instead of the project's root `.markdownlint-cli2.cjs`?